src/sinks/humio_logs.rs

```diff
@@ -23,15 +33,22 @@ inventory::submit! {
     SinkDescription::new_without_default::<HumioLogsConfig>("humio_logs")
 }
 
+fn default_encoding() -> EncodingConfigWithDefault<Encoding> {
+    EncodingConfigWithDefault::from(Encoding::Json)
+}
+
 #[typetag::serde(name = "humio_logs")]
 impl SinkConfig for HumioLogsConfig {
     fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        if self.encoding.codec != Encoding::Json {
```

Comment:
I wonder if there is a better way around this, in theory could we just implement a custom encoding enum here that doesn't have `Json`?

Reply:
Yeah I don't think this is the nicest way either but it's very straightforward

Reply:
The one thing that worries me is I've seen serde sometimes enumerate all the possible options but in this case a user would see json as being available but would fail when you configure the sink. Just something we should check.

Reply:
Yeah, hm... The only way to fix that would be to give this a custom encoding time and transmute it

Reply:
Why not write a custom enum here that doesn't include json?

Reply:
![](https://media.giphy.com/media/l36kU80xPf0ojG0Erg/giphy-downsized.gif)